sympy/polys/rings.py

```diff
@@ -2217,6 +2219,39 @@ def _gcd_QQ(self, g):
 
         return h, cff, cfg
 
+    def _gcd_prs(self, g):
+        """Helper function for ``gcd_prs`` method."""
+
+        f = self
+        K = f.ring.domain
+
+        if not K.is_Exact:
+            K_exact = K.get_exact()
+
+            ring_approx = f.ring
+            ring_exact = K_exact[ring_approx.symbols].ring
+
+            f = f.set_ring(ring_exact)
+            g = g.set_ring(ring_exact)
+
+            h, cff, cfg = f._gcd_prs(g)
+
+            h = h.set_ring(ring_approx)
+            cff = cff.set_ring(ring_approx)
+            cfg = cfg.set_ring(ring_approx)
+
+            return h, cff, cfg
+        elif K.is_Field:
+            if K.is_QQ:
+                return f._gcd_QQ(g)
+
+            return gcd_prs(f, g)
+        else:
+            if K.is_ZZ:
+                return f._gcd_ZZ(g)
+
+            return gcd_prs(f, g)
```

Comment:
Why is this logic here? This function is called `_gcd_prs` so it should be the PRS algorithm but here for ZZ and QQ the heugcd method will be used.